core/logger/splunk.go

```diff
@@ -0,0 +1,152 @@
+package logger
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+)
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type event struct {
+	Time       int64           `json:"time"`                 // epoch time in nano-seconds
+	Host       string          `json:"host"`                 // hostname
+	Source     string          `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
+	SourceType string          `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
+	Index      string          `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
+	Event      json.RawMessage `json:"event"`                // zap json logger
+}
+
+type splunkSink struct {
+	HTTPClient    httpClient
+	nowFn         func() int64
+	URL           string // URL to Splunk HTTP Event Collector
+	Hostname      string // Hostname is the human-readable machine name sending logs
+	Token         string // Splunk HTTP Event Collector token
+	Source        string // Splunk source field value, description of the source of the event
+	SourceType    string // Splunk source type, optional name of a sourcetype field value
+	Index         string // Splunk Index, optional name of the Splunk index to store the event in
+	SkipTLSVerify bool   // Skip verifying the certificate of the HTTP Event Collector
+	events        [][]byte
+	sync.Mutex
+}
+
+func envVarOrDefault(key string, defaultValue string) string {
+	envVar := os.Getenv(key)
+	if envVar == "" {
+		return defaultValue
+	}
+	return envVar
+}
+
+func initialize(s *splunkSink) *splunkSink {
+	tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: s.SkipTLSVerify}}
+	s.HTTPClient = &http.Client{Timeout: time.Second * 20, Transport: tr}
```

Comment:
Is it not possible to use the global http client from `utils/http.go`?

Reply:
No, because we want to override TLS settings with the `InsecureSkipVerify` and set a timeout. Here is a bit of background: https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779